code/game/machinery/vending/cm_vending.dm

```diff
@@ -443,7 +443,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 //------------TGUI PROCS---------------
 
 /obj/structure/machinery/cm_vending/ui_data(mob/user)
-	return vendor_user_ui_data(src, user)
+	return vendor_user_ui_data(user)
```

Comment:
oh wait this proc should just return what vendor_user_ui_data and we should get rid of that proc if it's not used anywhere else